src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -266,20 +274,42 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
         $fragment = isset($extra['_fragment']) ? $extra['_fragment'] : '';
         unset($extra['_fragment']);
 
-        if ($extra && $query = http_build_query($extra, '', '&')) {
-            // "/" and "?" can be left decoded for better user experience, see
+        // ignore null values
+        $extra = array_filter($extra, function($value) { return isset($value); });
```

Comment:
`isset` seems to be hacky when the var is clearly defined – if u want to check for null, compare it with null:

`$extra = array_filter($extra, function($value) { return null === $value; });`